### PR TITLE
fix: aggregated measure data result should only contain one serie element

### PR DESCRIPTION
**Pull request by MadsDue**
## Description

aggregated measure data result should only contain one serie element with the latest calculated TimeSeriePoints. 

Then we have split CalculationResultQueries into two classes. The original and a new one, which is responsble for query calculation result triggered by the aggregated measure data request. 

## References

* #0000

### CODE REVIEW

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/RequestCalculationResult/RequestCalculationResultQueries.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    private static TimeSeriesPoint CreateTimeSeriesPoint(SqlResultRow row)
+    {
+        var time = SqlResultValueConverters.ToDateTimeOffset(row[EnergyResultColumnNames.Time])!.Value;
+        var quantity = SqlResultValueConverters.ToDecimal(row[EnergyResultColumnNames.Quantity])!.Value;
+        var quality = SqlResultValueConverters.ToQuantityQuality(row[EnergyResultColumnNames.QuantityQuality]);
+        return new TimeSeriesPoint(time, quantity, quality);
+    }
+
+    private static EnergyResult CreateEnergyResult(
```
Comment (JesperGomes) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/RequestCalculationResult/RequestCalculationResultQueries.cs:107]: Wouldn't it make sense to re-use this (and CreateTimeSeriesPoint) for the two query classes?

File: source/dotnet/wholesale-api/Events/Events.Infrastructure/InboxEvents/AggregatedTimeSeriesRequestAccepted.proto [hunk 2]
```diff
@@ -19,7 +19,7 @@ import "google/protobuf/timestamp.proto";
 option csharp_namespace = "Energinet.DataHub.Edi.Responses";
 
 message AggregatedTimeSeriesRequestAccepted {
-  repeated Serie series = 1;
+  Serie serie = 1;
```
Comment (JesperGomes) [inline source/dotnet/wholesale-api/Events/Events.Infrastructure/InboxEvents/AggregatedTimeSeriesRequestAccepted.proto:22]: Just curious - why not call it TimeSeries?

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/RequestCalculationResult/RequestCalculationResultQueries.cs (hunk 1 above)
Comment (MadsDue) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/RequestCalculationResult/RequestCalculationResultQueries.cs]: I have added a factory for this :D

File: source/dotnet/wholesale-api/Events/Events.Infrastructure/InboxEvents/AggregatedTimeSeriesRequestAccepted.proto (hunk 2 above)
Comment (JesperGomes) [reply source/dotnet/wholesale-api/Events/Events.Infrastructure/InboxEvents/AggregatedTimeSeriesRequestAccepted.proto:22]: And what happened to the 's' at the end :-)

Comment (MadsDue) [reply source/dotnet/wholesale-api/Events/Events.Infrastructure/InboxEvents/AggregatedTimeSeriesRequestAccepted.proto:22]: Good point :D

File: source/dotnet/wholesale-api/Events/Events.UnitTests/InboxEvents/AggregatedTimeSeriesMessageFactoryTests.cs [hunk 3]
```diff
@@ -48,22 +48,20 @@ public void Create_WithCalculationResultFromTotalProductionPerGridArea_CreatesAc
...
         var responseBody = AggregatedTimeSeriesRequestAccepted.Parser.ParseFrom(response.Body);
-        Assert.All(responseBody.Series, serie =>
-        {
-            Assert.Equal(_gridArea, serie.GridArea);
-            Assert.Equal(Energinet.DataHub.Edi.Responses.TimeSeriesType.Production, serie.TimeSeriesType);
-            Assert.Equal(new Timestamp() { Seconds = _periodStart.ToUnixTimeSeconds() }, serie.Period.StartOfPeriod);
-            Assert.Equal(new Timestamp() { Seconds = _periodEnd.ToUnixTimeSeconds() }, serie.Period.EndOfPeriod);
-            Assert.Equal(energyResult.TimeSeriesPoints.Length, serie.TimeSeriesPoints.Count);
-        });
+
```
Comment (JesperGomes) [inline source/dotnet/wholesale-api/Events/Events.UnitTests/InboxEvents/AggregatedTimeSeriesMessageFactoryTests.cs:59]: AssertionScope?

Comment (MadsDue) [reply source/dotnet/wholesale-api/Events/Events.UnitTests/InboxEvents/AggregatedTimeSeriesMessageFactoryTests.cs:59]: Do you mean fluent Assertion?

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/RequestCalculationResult/RequestCalculationResultQueries.cs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    public static string[] SqlColumnNames { get; } =
+    {
+        EnergyResultColumnNames.BatchId, EnergyResultColumnNames.GridArea, EnergyResultColumnNames.FromGridArea,
+        EnergyResultColumnNames.TimeSeriesType, EnergyResultColumnNames.EnergySupplierId,
+        EnergyResultColumnNames.BalanceResponsibleId, EnergyResultColumnNames.Time,
+        EnergyResultColumnNames.Quantity, EnergyResultColumnNames.QuantityQuality,
+        EnergyResultColumnNames.CalculationResultId, EnergyResultColumnNames.BatchProcessType,
+    };
+
+    private static TimeSeriesPoint CreateTimeSeriesPoint(SqlResultRow row)
```
Comment (JesperGomes) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/RequestCalculationResult/RequestCalculationResultQueries.cs:100]: Shared factory?

**Review by JesperGomes — DISMISSED** [review]

**Review by JesperGomes — APPROVED** [review]

File: source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/RequestCalculationResult/RequestCalculationResultQueries.cs [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+using Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.SqlStatements.Mappers;
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.CalculationResults;
+using Energinet.DataHub.Wholesale.CalculationResults.Interfaces.CalculationResults.Model;
+using Energinet.DataHub.Wholesale.Common.Databricks.Options;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Energinet.DataHub.Wholesale.CalculationResults.Infrastructure.RequestCalculationResult;
+
+public class RequestCalculationResultQueries : IRequestCalculationResultQueries
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/RequestCalculationResult/RequestCalculationResultQueries.cs:28]: This class seems to not be covered by (any?) tests.

Comment (MadsDue) [reply source/dotnet/wholesale-api/CalculationResults/CalculationResults.Infrastructure/RequestCalculationResult/RequestCalculationResultQueries.cs:28]: There is this one: https://github.com/Energinet-DataHub/opengeh-wholesale/blob/main/source/dotnet/wholesale-api/CalculationResults/CalculationResults.IntegrationTests/Infrastructure/RequestCalculationResult/RequestCalculationResultQueriesTests.cs

**Pull request merged**